src/main/kotlin/org/rust/lang/core/completion/RsCommonCompletionProvider.kt

```diff
@@ -156,3 +230,26 @@ private fun filterMethodCompletionVariantsByTraitBounds(
         return false
     }
 }
+
+private fun addProcessedPathName(
+    processor: RsResolveProcessor,
+    processedPathNames: MutableSet<String>
+): RsResolveProcessor = fun(it: ScopeEntry): Boolean {
+    if (it.element != null) {
+        processedPathNames.add(it.name)
+    }
+    return processor(it)
+}
+
+private fun addImport(context: InsertionContext, candidate: ImportCandidate, mod: RsMod) {
+    val project = context.project
+    val file = context.file
+
+    val manager = PsiDocumentManager.getInstance(project)
+    val document = manager.getDocument(file)
+    if (document != null) {
+        manager.commitDocument(document)
+    }
```

Comment:
Is it really needed here?
Looks like it's called in `com.intellij.codeInsight.completion.CodeCompletionHandlerBase#insertItem` where all documents are commited before insert handler invocation